File: EclipseWorkspace/ScalaServer/src/misc.scala
object misc
{
  class FloatArray(width:Int, height:Int)
  {
    private var myArray = Array.ofDim[Float](width,height)
    def Width():Int = width
    def Height():Int = height
    def Get(x:Int, y:Int):Float = myArray(x)(y)
    def Set(x:Int, y:Int, value:Float) = myArray(x)(y) = value
    def Map(fn:(Float)=>(Float)) = myArray.map(i:Array[Float] => fn(i))
  }
    
  class IntArray(width:Int, height:Int)
  {
    private var myArray = Array.ofDim[Int](width,height)    
    def Width():Int = width
    def Height():Int = height
    def Get(x:Int, y:Int):Int = myArray(x)(y)
    def Set(x:Int, y:Int, value:Int) = myArray(x)(y) = value
  }
}

File: DotNet/WorldServer/World/Generator/Misc.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace Sean.World
{
    public static class Misc
    {
        public static T[][] GetEmptyArray<T>(int width, int height)
        {
            var image = new T[width][];

            for (int i = 0; i < width; i++)
            {
                image[i] = new T[height];
            }
            return image;
        }

        private static byte[] GenerateDeterministicHash(int x, int y, int z, int worldSeed)
        {
            using (MD5 md5Hash = MD5.Create ()) 
            {
                string input = $"{x}-{y}-{z}-{worldSeed}";
                // Convert the input string to a byte array and compute the hash.
                return md5Hash.ComputeHash(Encoding.UTF8.GetBytes(input));
            }
        }
        private static byte[] GenerateDeterministicHash(int x, int z, int worldSeed)
        {
            using (MD5 md5Hash = MD5.Create ()) 
            {
                string input = $"{x}-{z}-{worldSeed}";
                // Convert the input string to a byte array and compute the hash.
                return md5Hash.ComputeHash(Encoding.UTF8.GetBytes(input));
            }
        }
        public static float GetDeterministicFloat (int x, int z, int worldSeed)
        {
            byte[] data = GenerateDeterministicHash(x,z,worldSeed);
            return System.BitConverter.ToSingle(data, 0);
        }
        public static uint GetDeterministicInt (int x, int y, int z, int worldSeed)
        {
            byte[] data = GenerateDeterministicHash(x,y,z,worldSeed);
            return System.BitConverter.ToUInt32(data, 0);
        }
        public static uint GetDeterministicInt (int x, int z, int worldSeed)
        {
            byte[] data = GenerateDeterministicHash(x,z,worldSeed);
            return System.BitConverter.ToUInt32(data, 0);
        }

    }
}

